### PR TITLE
`SymbolicOp`s are queued correctly with `qml.cond`

### PR DESCRIPTION
**Pull request by mudit2812**
**Context:**
Any function that is conditioned with `qml.cond` and takes operators as arguments was being queued incorrectly. This is especially problematic for op_math functions, which generally allow operators as input, not just callables.

**Description of the Change:**
Added check to remove any operators from the queue that are present in the arguments to the conditioned function.

I initially thought about also checking for lists of operators, but that seems like a rabbit hole (why not also check for lists of lists of lists ... of operators). I also don't want to encourage users to create their own qfuncs that accept operators. This bug only exists because we have pennylane functions that accept operators, but creating qfuncs that accept operators will cause problems with queuing even outside `qml.cond` and should be avoided.

**Benefits:**
`op_math` functions work correctly with `qml.cond`.

**Possible Drawbacks:**

**Related GitHub Issues:**
#4822

### CODE REVIEW

**Comment by mudit2812** [comment]
[sc-49921]

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -542,6 +542,9 @@
 
 <h3>Bug fixes 🐛</h3>
 
+`qml.cond` no longer incorrectly queues operators used as qfunc arguments.
```
Comment (lillian542) [inline doc/releases/changelog-dev.md:545]:
```suggestion
* `qml.cond` no longer incorrectly queues operators used as qfunc arguments.
```

**Review by lillian542 — APPROVED** [review]
Looks good to me 🚀

**Review by timmysilv — APPROVED** [review]
👢
⬇️
🐛

**Pull request merged**